### ISSUE TITLE
About  fine tuning superpoint model by using own warped_pair

### ISSUE THREAD

**Issue by nkuwangfeng** [rpautrat/SuperPoint#188]
Good work!!!
Hello, I want to train a more robust descriptor to illumination. I see that in the training of super point descriptor, a pair of homography transformation images are needed to extract features and calculate the descriptor loss. Now I have an idea. I'd like to ask you how you need to modify it to train a more robust descriptor to light.

1. Collect image data of different light intensities in the same scene.

2. Using the official trained superpoint to extract features, using the outdoor weight model of Superglue to complete the feature matching, we can get the homography transformation matrix H of the two images, and the position of the feature points as the true value of the label.

3. Put the two pictures into the superpoint training as image and warped_ pair.Because we already know the true value of the point, and the H matrix contains the true value of the descriptor, can we train a more robust super point model to illumination.

 I'm looking forward to your reply.Thanks!

**Comment by rpautrat** [comment]
Hi,

Extending the descriptor training pipeline to something more robust to illumination is definitely possible.

Regarding your proposal, if you take image data from different viewpoint with a camera, then they will not be related by a homography anymore (unless the scene is completely planar, as in HPatches). If the scene is not planar, you need the depth values of each pixel and the camera matrices (intrinsics and extrinsics) to find the correspondence between two views.

Another option is to take the image data with a fixed camera and varying illumination, and to warp the image later during training by a homography to create the change in viewpoint (as is currently been done in the COCO training). Such datasets with fixed viewpoint already exist, for example [here](https://projects.csail.mit.edu/illumination/) and [here](https://github.com/majedelhelou/VIDIT) (this one is synthetic). But by acquiring them yourself, you can certainly ensure a larger diversity of illumination and scenes.

**Comment by rpautrat** [comment]
I am closing this now, feel free to reopen it if you still have questions.

**Comment by zzkslam** [comment]
I also want to train a more robust descriptor to illumination and I want to use the first link of dataset(A Dataset of Multi-Illumination Images in the Wild) you provide.Can you give me some advice should I use this dataset to retrain the magicpoint or retrain the superpoint ,or both of them?

**Comment by rpautrat** [comment]
I think that you can use the pre-trained model of SuperPoint on COCO, and directly fine-tune it on the multi illumination dataset. This should work and would be the fastest way to get your robust descriptor.

**Comment by zzkslam** [comment]
OK,thank you very much for your advice.
